src/web/security/core/src/main/java/org/geoserver/security/web/url/RegexCheckPage.java

```diff
@@ -90,10 +90,34 @@ private void initUI(IModel<RegexURLCheck> bean) {
         add(form);
     }
 
+    /**
+     * Generate an example reglar expression.
+     *
+     * <p>This example processes {@link #getOWSURL()} to demonstrate:
+     *
+     * <ul>
+     *   <li>Shows {@code https?:} to allow support both http and https
+     *   <li>Escaping {@code \.} characters
+     *   <li>Disallow any {@code ../} relative paths
+     *   <li>Allow anything after {@code ?} query
+     * </ul>
+     *
+     * @return example regular expression
+     */
     private String getDefaultRegex() {
-        return "^" + getOWSURL() + ".*$";
+        String url = getOWSURL();
+
+        url = url.replace("http:", "https?:");
+        url = url.replace(".", "\\.");
+
+        return "^" + url + "((?!\\.\\./).)*(\\?.*)?$";
```

Comment:
It seems this is a match ``http://geoserver.org/test/..`` but I think it should not be

Reply:
The initial URL is ``http://localhot:8080/geoserver/ows`` so this turns into ``^https?://localhot:8080/geoserver/ows((?!\\.\\./).)*(\\?.*)?$``

So it should not match ``http://geoserver.org/test/..`` ?

Reply:
You're being too literal here, I just typed a test regex and a test string in regex101.com, in Java mode.
``^https?://localhost:8080/geoserver/ows((?!\\.\\./).)*(\\?.*)?$```
matches:
``http://localhost:8080/geoserver/ows/..``

A trailing ``..`` without a "/" after it is allowed. I don't think it should be.

See here for a playground with the above regex and exception: https://regex101.com/r/cHMq3W/1

Reply:
Good point!

If I understand the consequence of this ... if a normalization is available it should always be used. So that ``http://localhost:8080/geoserver/ows/..`` normalizes to ``http://localhost:8080/geoserver`` which would indeed fail checks.

Reply:
I covered this case in the geotools test cases for https://github.com/geotools/geotools/pull/4337 to ensure it is covered. 

Thanks for thinking through this awkward corner case.